src/database/mod.rs

```diff
@@ -349,12 +357,25 @@ impl Coin {
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub enum CoinType {
-    All,
-    Unspent,
+pub enum CoinStatus {
+    Unconfirmed,
+    Confirmed,
+    Spending,
```

Comment:
Let's have a doc comment for each variant about how they are mutually exclusive?

Reply:
I've now added a comment for each variant.